File: crates/sargon-uniffi/src/profile/supporting_types/account_for_display.rs
use crate::prelude::*;
use sargon::AccountForDisplay as InternalAccountForDisplay;

/// A minimal version of an [`Account`] meant for
/// display purposes within wallet
#[derive(Clone, PartialEq, Hash, Eq, InternalConversionV2, uniffi::Record)]
pub struct AccountForDisplay {
    pub address: AccountAddress,

    pub display_name: DisplayName,

    pub appearance_id: AppearanceID,
}

#[uniffi::export]
pub fn new_account_for_display_sample() -> AccountForDisplay {
    InternalAccountForDisplay::sample().into()
}

#[uniffi::export]
pub fn new_account_for_display_sample_other() -> AccountForDisplay {
    InternalAccountForDisplay::sample_other().into()
}

#[uniffi::export]
pub fn new_account_for_display_from_account(
    account: Account,
) -> AccountForDisplay {
    InternalAccountForDisplay::from(account.into_internal()).into()
}

File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/AccountForDisplay.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.Account
import com.radixdlt.sargon.AccountForDisplay
import com.radixdlt.sargon.newAccountForDisplayFromAccount

fun AccountForDisplay.Companion.from(account: Account) = newAccountForDisplayFromAccount(
    account = account
)